x-pack/plugins/lens/common/expressions/datatable/transpose_helpers.ts

```diff
@@ -86,6 +85,11 @@ export function transposeTable(
         transposedColumnId,
         metricsColumnArgs
       );
+
+      const colOrderMap = new Map(args.columns.map((c, i) => [c.columnId, i]));
+      firstTable.columns.sort((a, b) => {
+        return (colOrderMap.get(a.id) ?? 0) - (colOrderMap.get(b.id) ?? 0);
+      });
```

Comment:
Aligns the table column order to that of the args. This is to done to preserve the order of transposed columns as described in XXX.